internal/guest/bridge/bridge.go

```diff
@@ -365,7 +365,12 @@ func (b *Bridge) ListenAndServe(bridgeIn io.ReadCloser, bridgeOut io.WriteCloser
 	// Process each bridge response sync. This channel is for request/response and publish workflows.
 	go func() {
 		var resperr error
+		var err error
 		for resp := range b.responseChan {
+			_, span := oc.StartSpan(context.Background(), "request write response", oc.WithClientSpanKind)
+			defer span.End()
+			defer func() { oc.SetSpanStatus(span, err) }()
```

Comment:
these defer's wont run until the `go func()` ends, not at the end of each loop iteration, and the `err` variable will likely get overridden